core/src/Platforms/iOS/EmbeddedWebview/AuthenticationAgentUIViewController.cs

```diff
@@ -64,90 +63,93 @@ public override void ViewDidLoad()
 
             View.BackgroundColor = UIColor.White;
 
-            webView = new UIWebView(View.Bounds);
-            webView.ShouldStartLoad = (wView, request, navType) =>
+            WKWebViewConfiguration webConfiguration = new WKWebViewConfiguration();
+
+            wkWebView = new WKWebView(View.Bounds, webConfiguration);
+            NSUrl url = wkWebView.Url;
+
+            wkWebView.LoadRequest(new NSUrlRequest(url));
+
+            if (LoadView(url) == true)
             {
-                if (request == null)
-                {
-                    return true;
-                }
+                wkWebView.StopLoading();
 
-                string requestUrlString = request.Url.ToString();
+                // If the title is too long, iOS automatically truncates it and adds ...
+                this.Title = wkWebView.EvaluateJavaScriptAsync(@"document.title").ToString();
```

Comment:
EvaluateJavaScriptAsync returns a Task<string>, so calling toString() on it will return the Task.ToString() implementation, which is "System.Threading.Tasks.Task`1[System.String]"

What you want to do is to either: 

a) await the result
b) Wait(), potentially with a timeout
c) Use the non-async version, if there is one.